libcsl_godot/src/lib.rs

```diff
@@ -301,23 +336,94 @@ impl GTxBuilder {
         let amount_builder = output_builder
             .with_address(&address.bind().address)
             .next()
-            .expect("Error to build transaction output");
-        let output = amount_builder
-            .with_coin_and_asset(
-                &coin
-                    .bind()
-                    .b
-                    .as_u64()
-                    .expect("Output lovelace exceeds maximum"),
-                &multiasset_from_dictionary(&assets),
-            )
-            .build()
-            .expect("Error to build amount output");
+            .expect("Failed to build transaction output");
+        let output = 
+            if coin.bind().gt(BigInt::zero()) {
+                amount_builder
+                    .with_coin_and_asset(
+                        &coin
+                            .bind()
+                            .b
+                            .as_u64()
+                            .expect("Output lovelace exceeds maximum"),
+                        &multiasset_from_dictionary(&assets),
+                    )
+                    .build()
+                    .expect("Failed to build amount output")
+            } else {
+                amount_builder
+                    .with_asset_and_min_required_coin_by_utxo_cost(
+                        &multiasset_from_dictionary(&assets),
+                        &CSL::DataCost::new_coins_per_byte(&to_bignum(self.protocol_parameters.coins_per_utxo_byte))
+                    )
+                    .expect("Failed to build minUTxO output")
+                    .build()
+                    .expect("Failed to build amount output")
+            };
+
         self.tx_builder
             .add_output(&output)
             .expect("Could not add output");
     }
 
+    #[func]
+    fn mint_assets(
+        &mut self,
+        script: Gd<PlutusScript>,
+        tokens: Dictionary,
+        redeemer: PackedByteArray,
+    ) {
+        let bound = script.bind();
+        let script = &bound.deref().script;
+        let mut index: u32 = 0;
+        let num_scripts: u32 = self.plutus_scripts.len() as u32;
+        while index < num_scripts {
```

Comment:
Are we using a while loop here instead of a standard for loop because `PlutusScript` does not implement the iterator trait? Is there a workaround to avoid this?

Reply:
Yes, I think that is the case.